linode/manager
# refactor: Refactor Cypress test setup, cache API account request

## Description 📝
This is a first pass at fixing some tests that have been flaky recently by restoring the account request mock using real API data. One Jenkins test account is consistently taking 30+ seconds to make requests to the `/account` endpoint which is causing random failures and making it hard to pinpoint flake caused by more specific issues.

I plan to follow up with a more comprehensive PR containing more targeted fixes for the rescue, rebuild, and resize test flake we've been seeing (I originally intended for that to be _this_ PR), but want to prioritize this fix since it'll go a long way towards getting our Jenkins queue under control.

## Major Changes 🔄
- Add missing clean up to some domains tests to resolve resource accumulation issues
- Restore account request mocking by using real account response data rather than factory data
  - This should improve performance without causing subtle and hard-to-troubleshoot issues like what cropped up during the DBaaS EOS changes
- Refactor code to set up Cypress tests (adding commands, setting up request tracking, etc.), moving much of that code into a new `support/setup` dir
- Merge OAuth tests into a single spec

## How to test 🧪
This is targeting a CI-specific issue, so we will have to rely on the Jenkins run to see if this makes any impact. Specifically, one runner in particular is consistently taking 45+ minutes even when all of the tests pass; we can validate that this PR resolves that issue by confirming that all 4 runners run in a reasonable length of time (~20-25 mins)
